src/org/usfirst/frc/team930/robot/subsystems/Drivetrain.java

```diff
@@ -1,18 +1,220 @@
+		this.setMotors();
+	}
+
+	public void setMotors() {
+
+		frDrive = new CANTalon(FRONT_RIGHT);
+		flDrive = new CANTalon(FRONT_LEFT);
+		blDrive = new CANTalon(BACK_LEFT);
+		brDrive = new CANTalon(BACK_RIGHT);
+
+		frRot = new CANJaguar(FRONT_RIGHT);
+		flRot = new CANJaguar(FRONT_LEFT);
+		blRot = new CANJaguar(BACK_LEFT);
+		brRot = new CANJaguar(BACK_RIGHT);
+
+		gyro = new Gyro(1);
+
+		frRot.setPositionMode(CANJaguar.kQuadEncoder, CODES_PER_REV,
+				DEFAULT_JAG_P, DEFAULT_JAG_I, 0);
+		flRot.setPositionMode(CANJaguar.kQuadEncoder, CODES_PER_REV,
+				DEFAULT_JAG_P, DEFAULT_JAG_I, 0);
+		blRot.setPositionMode(CANJaguar.kQuadEncoder, CODES_PER_REV,
+				DEFAULT_JAG_P, DEFAULT_JAG_I, 0);
+		brRot.setPositionMode(CANJaguar.kQuadEncoder, CODES_PER_REV,
+				DEFAULT_JAG_P, DEFAULT_JAG_I, 0);
+
+		frDrive.changeControlMode(ControlMode.Speed);
+		frDrive.setPID(DEFAULT_TAL_P, DEFAULT_TAL_I, 0);
+		flDrive.changeControlMode(ControlMode.Speed);
+		flDrive.setPID(DEFAULT_TAL_P, DEFAULT_TAL_I, 0);
+		blDrive.changeControlMode(ControlMode.Speed);
+		blDrive.setPID(DEFAULT_TAL_P, DEFAULT_TAL_I, 0);
+		brDrive.changeControlMode(ControlMode.Speed);
+		brDrive.setPID(DEFAULT_TAL_P, DEFAULT_TAL_I, 0);
+
+		frRot.enableControl();
+		flRot.enableControl();
+		blRot.enableControl();
+		brRot.enableControl();
+
+		gyro.initGyro();
+	}
+
+	public void drive(double forward, double strafe, double rot) {
+
+		//
+		// if (forward == 0 && strafe == 0) {
+		// frRot.setI(0);
+		// flRot.setI(0);
+		// blRot.setI(0);
+		// brRot.setI(0);
+		// } else {
+		// frRot.setI(DEFAULT_I);
+		// flRot.setI(DEFAULT_I);
+		// blRot.setI(DEFAULT_I);
+		// brRot.setI(DEFAULT_I);
+		// }
+
+		// double heading = Math.PI / 2 - gyro.getAngle() * (Math.PI / 180);
+		swerve.updateSwerve(forward, strafe, rot);
+
+		System.out.println("fr encoder " + frRot.getPosition());
+
+		this.quickAngle(swerve.output(Outputs.frontRightAngle),
+				swerve.output(Outputs.frontRightSpeed), frRot, frDrive);
+		this.quickAngle(swerve.output(Outputs.frontLeftAngle),
+				swerve.output(Outputs.frontLeftSpeed), flRot, flDrive);
+		this.quickAngle(swerve.output(Outputs.backLeftAngle),
+				swerve.output(Outputs.backLeftSpeed), blRot, blDrive);
+		this.quickAngle(swerve.output(Outputs.backRightAngle),
+				swerve.output(Outputs.backRightSpeed), brRot, brDrive);
+
+		// SmartDashboard.putNumber("front right", frRot.getFaults());
+		// SmartDashboard.putNumber("front left", flRot.getFaults());
+		// SmartDashboard.putNumber("back right", brRot.getFaults());
+		// SmartDashboard.putNumber("back left", blRot.getFaults());
+		//
+
+		/*
+		 * SmartDashboard.putNumber("Front right encoder", frRot.getPosition());
+		 * SmartDashboard.putNumber("Front left encoder", flRot.getPosition());
+		 * SmartDashboard.putNumber("Back right encoder", brRot.getPosition());
+		 * SmartDashboard.putNumber("Back left encoder", blRot.getPosition());
+		 */
+
+		SmartDashboard.putNumber("FR angle output",
```

Comment:
On competition robot, we should probably remove all useless SmartDashboard calls. I believe this is tested enough that we do not need these ones any more.